### ISSUE TITLE
Suddenly cannot publish both Wheel and Source code to PyPi

### ISSUE THREAD

**Issue by RobertoPrevato** [pypa/warehouse#6700]
**Describe the bug**
Up to three days ago (at least), when publishing distribution packages to PyPi, it was possible to publish both a Wheel and Source code ([example here](https://pypi.org/project/essentials/1.0.9/#files)). Today, publishing both Wheel and source code fails, with the following response from server (see [here a public pipeline in Azure DevOps](https://dev.azure.com/robertoprevato/Nest/_releaseProgress?_a=release-environment-logs&releaseId=11&environmentId=22)):

```bash
2019-09-24T15:33:39.4707295Z Uploading essentials-1.1.0-py3-none-any.whl
2019-09-24T15:33:39.4707705Z 
2019-09-24T15:33:39.4708057Z   0%|          | 0.00/15.7k [00:00<?, ?B/s]
2019-09-24T15:33:39.8438918Z  51%|█████     | 8.00k/15.7k [00:00<00:00, 28.2kB/s]
2019-09-24T15:33:39.8504065Z 100%|██████████| 15.7k/15.7k [00:00<00:00, 18.0kB/s]
2019-09-24T15:33:39.8504885Z Uploading essentials-1.1.0.tar.gz
2019-09-24T15:33:39.8505241Z 
2019-09-24T15:33:40.6700964Z   0%|          | 0.00/13.0k [00:00<?, ?B/s]
2019-09-24T15:33:40.6727491Z 100%|██████████| 13.0k/13.0k [00:00<00:00, 16.3kB/s]HTTPError: 400 Client Error: This filename has already been used, use a different version. See https://pypi.org/help/#file-name-reuse for url: https://upload.pypi.org/legacy/
2019-09-24T15:33:40.6727752Z 
2019-09-24T15:33:40.6728102Z NOTE: Try --verbose to see response content.
```

So, only the Wheel is available in PyPi ([example](https://pypi.org/project/essentials/1.1.0/#files)).

**Note**: it is still possible to publish both Wheel and source code to `test.pypi.org` ([example](https://test.pypi.org/project/essentials/1.1.0/#files)).

**Expected behavior**
Should it be still possible to publish both a Wheel and source code packages in a tar.gz file?

**Comment by ewdurbin** [comment]
Hello @RobertoPrevato, this is actually due to previous use of this project name on PyPI from 2018-02-24  through 2018-11-22. The owner of the project at the time released essential 1.1.0 with only one release file `essentials-1.1.0.tar.gz`. The project and release were later removed.

PyPI does not allow reuse of file names after a release file, release, or project are removed.

So the issue that you're running into is a collision with a previously used filename.

**Comment by RobertoPrevato** [comment]
Hi @ewdurbin, thank You for your kind explanation. It didn't occurred to me that this could be the case! Everything's clear now. It looks a bit weird, though, that I could upload the Wheel (only the .tar.gz file with source code failed uploading).

Everything's clear. :+1:
